Source/BlueprintSearchVSExtension/Source/Commands/CommandHelpers/PathFinderHelper.cs

```diff
@@ -0,0 +1,89 @@
+//Copyright(C) Coconut Lizard Limited. All rights reserved.
+
+// ---------------------------------------------------------
+
+using EnvDTE;
+using EnvDTE80;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Windows;
+
+namespace BlueprintSearch.Commands.CommandHelpers
+{
+	public static class PathFinderHelper
+	{
+		public static DTE2 DTEService;
+
+		public static string UProjectFilePath { get; set; }
+
+		public static string UECommandLineFilePath { get; set; }
+
+		public static void FindPaths()
+		{
+			List<Project> ProjectsList = new List<Project>();
+			ProjectsList.AddRange(DTEService.Solution.Projects.Cast<Project>());
+
+			for (int i = 0; i < ProjectsList.Count; i++)
+			{
+				ProjectsList.AddRange(ProjectsList[i].ProjectItems.Cast<ProjectItem>().Select(x => x.SubProject).OfType<Project>());
+			}
+
+			for (int i = 0; i < ProjectsList.Count; ++i)
```

Comment:
Should this be a foreach rather than an index based loop?

Reply:
For reference if you've unfamiliar with C# foreach statements :)
Var name has to be some permutation of Proj/project (but not 'Project', this is why the unreal coding standard has prefix's on class names)
```suggestion
			foreach (Project project in ProjectsList)
```